Okta.Xamarin/Okta.Xamarin/Base64UrlEncoder.cs

```diff
@@ -30,109 +30,109 @@
-			s = s.Replace(base64Character63, _base64UrlCharacter63);  // 63rd char of encoding
+        /// <summary>
+        /// Converts a subset of an array of 8-bit unsigned integers to its equivalent string representation that is encoded with base-64-url digits. Parameters specify
+        /// the subset as an offset in the input array, and the number of elements in the array to convert.
+        /// </summary>
+        /// <param name="inArray">An array of 8-bit unsigned integers.</param>
+        /// <returns>The string representation in base 64 url encodingof length elements of inArray, starting at position offset.</returns>
+        /// <exception cref="ArgumentNullException">'inArray' is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">offset or length is negative OR offset plus length is greater than the length of inArray.</exception>
+        public static string Encode(byte[] inArray)
+        {
+            string s = Convert.ToBase64String(inArray, 0, inArray.Length);
+            s = s.Split(base64PadCharacter)[0]; // Remove any trailing padding
+            s = s.Replace(base64Character62, base64UrlCharacter62);  // 62nd char of encoding
+            s = s.Replace(base64Character63, _base64UrlCharacter63);  // 63rd char of encoding
 
-			return s;
-		}
+            return s;
+        }
 
-		/// <summary>
-		///  Converts the specified string, which encodes binary data as base-64-url digits, to an equivalent 8-bit unsigned integer array.</summary>
-		/// <param name="str">base64Url encoded string.</param>
-		/// <returns>UTF8 bytes.</returns>
-		public static byte[] DecodeBytes(string str)
-		{
+        /// <summary>
+        ///  Converts the specified string, which encodes binary data as base-64-url digits, to an equivalent 8-bit unsigned integer array.</summary>
+        /// <param name="str">base64Url encoded string.</param>
+        /// <returns>UTF8 bytes.</returns>
+        public static byte[] DecodeBytes(string str)
+        {
 
-			// 62nd char of encoding
-			str = str.Replace(base64UrlCharacter62, base64Character62);
+            // 62nd char of encoding
+            str = str.Replace(base64UrlCharacter62, base64Character62);
 
-			// 63rd char of encoding
-			str = str.Replace(_base64UrlCharacter63, base64Character63);
+            // 63rd char of encoding
+            str = str.Replace(_base64UrlCharacter63, base64Character63);
 
-			// check for padding
-			switch (str.Length % 4)
-			{
-				case 0:
-					// No pad chars in this case
-					break;
-				case 2:
-					// Two pad chars
-					str += doubleBase64PadCharacter;
-					break;
-				case 3:
-					// One pad char
-					str += base64PadCharacter;
-					break;
-			}
+            // check for padding
+            switch (str.Length % 4)
+            {
+                case 0:
+                    // No pad chars in this case
+                    break;
+                case 2:
```

Comment:
Maybe `case 1` is missed here.

Reply:
This file is legacy and should probably be replaced, at some point, with an established bas364 encoder.